src/include/optimizer/rewriter.h

```diff
@@ -0,0 +1,51 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// rewriter.h
+//
+// Identification: src/include/optimizer/rewriter.h
+//
+// Copyright (c) 2015-2018, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <memory>
+
+#include "expression/abstract_expression.h"
+#include "optimizer/optimizer_metadata.h"
+#include "optimizer/optimizer_task_pool.h"
+#include "optimizer/absexpr_expression.h"
+
+namespace peloton {
+namespace optimizer {
+
+class Rewriter {
+
+ public:
+  Rewriter(const Rewriter &) = delete;
+  Rewriter &operator=(const Rewriter &) = delete;
+  Rewriter(Rewriter &&) = delete;
+  Rewriter &operator=(Rewriter &&) = delete;
```

Comment:
Not sure if this is necessary as this is default behavior I think

Reply:
Consider DISALLOW_COPY_AND_MOVE macro.